### PR TITLE
MONEY-511

### PR DESCRIPTION
**Pull request by mtowers**
* make publishing connections single-threaded by default
* add manual connection failure handling
* update tests
* increment version number

### CODE REVIEW

File: lib/realself/stream/publisher.rb [hunk 1]
```diff
@@ -3,33 +3,63 @@
...
+        # default to single threded connections for publishing
+        # unless explicitly specified otherwise
+        @rmq_config[:threaded] = false unless config[:threaded]
+
+        initialize_connection
       end
 
       def publish(item, routing_key, content_type = 'application/json')
-        @channel.open unless @channel.open?
+        begin
```
Comment (wulftone) [inline lib/realself/stream/publisher.rb]: i believe you can un-nest one level by remove the `begin` and correlating `end`, but keep the `retry`

File: lib/realself/stream/publisher.rb [hunk 2]
```diff
@@ -3,33 +3,63 @@
...
+          tries ||= 1
```
Comment (edyesed) [inline lib/realself/stream/publisher.rb]: would it be worthwhile to fetch the X-death header instead of calculating `tries`  ?

**Comment by edyesed** [comment]
looks reasonable to me. TIL :ruby:  `retry`  :shipit:

**Comment by wulftone** [comment]
🚀 🏉 👟

**Pull request merged**